File: Practice(Beginner)/Python3/Smallest_Numbers_of_Notes.py
# Source: https://www.codechef.com/problems/FLOW005

n = int(input())
l = [100,50,10,5,2,1]
for it in range(n):
    p = int(input())
    c = 0
    while(p!=0):
        for i in l:
            if(i <= p):
                p = p - i
                c += 1
                break
    print(c)

File: Practice(Beginner)/C++/Smallest Numbers of Notes.cpp
// Source: https://www.codechef.com/problems/FLOW005

#include <bits/stdc++.h>
using namespace std;

int main() {

    int t, n;
    cin >> t;
    vector<int> v{100,50,10,5,2,1};
    for (int i = 0; i < t; ++i) {
        cin >> n;
        int count = 0;
        while(n){
            for (int j = 0; j <= 6; ++j) {
                if(v[j] <= n){
                    n -= v[j];
                    count++;
                    break;
                }
            }
        }
        cout << count << endl;
    }

    return 0;
}
